airflow/providers/sftp/hooks/sftp.py

```diff
@@ -45,9 +45,12 @@ class SFTPHook(SSHHook):
           permissions.
 
     Errors that may occur throughout but should be handled downstream.
+
+    :param sftp_conn_id: The :ref:`sftp connection id<howto/connection:sftp>`
+    :type sftp_conn_id: str
     """
 
-    conn_name_attr = 'ftp_conn_id'
+    conn_name_attr = 'sftp_conn_id'
```

Comment:
Wouldn’t think be backwards-incompatible? The change is the right thing to do, but may need a deprecation period?

Reply:
This change is wrong.

It should remain ftp_conn_id because this is the name of the constructor argument (below:
```
    def __init__(self, ftp_conn_id: str = 'sftp_default', *args, **kwargs) -> None:
```

It is later remapped to "ssh_conn_id" just before running the constructor of the parent class, so it does not matter. Note that it is just a name used when the connection is instantiated dynamically so it does not really matter what it is is, as long as it is consistent. And if someone ever created the Hook manually with keyword argument, changing it would mean backwards incompatibility as mentioned by @uranusjr

Reply:
@potiuk @uranusjr thanks for the review. I did not fully understand how this worked, I'll change it back!